### PR TITLE
chore: move `cy.type()` event tests to its own file.

### PR DESCRIPTION
**Pull request by sainthkh**
- Next step of #20067
- Related with [this feedback](https://github.com/cypress-io/cypress/pull/20067#discussion_r800713838)

### User facing changelog

N/A. It's just an internal cleanup.

### Additional details
- Why was this change necessary? => `type_spec.js` is too long. It takes about 2 minutes to finish. After adding event-related tests in this PR series, I thought it's a good idea to move them to its own file. 
- What is affected by this change? => N/A
- Any implementation details to explain? => Simply moving tests.

### How has the user experience changed?

N/A

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js [hunk 1]
```diff
@@ -0,0 +1,770 @@
...
+      cy.visit('fixtures/click-event-by-type.html')
+    })
+
+    describe('triggers', () => {
+      const targets = [
+        '#target-button-tag',
+        '#target-input-button',
+        '#target-input-image',
+        '#target-input-reset',
+        '#target-input-submit',
```
Comment (sainthkh) [inline packages/driver/cypress/integration/commands/actions/type_events_spec.js:588]: I added `#target` to make tests consistent with `cy.type(' ')` tests.

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js [hunk 2]
```diff
@@ -0,0 +1,770 @@
...
+      ]
+
+      targets.forEach((target) => {
+        it(target, () => {
+          cy.get(target).focus().type('{enter}')
+
+          cy.get('li').should('have.length', 4)
```
Comment (sainthkh) [inline packages/driver/cypress/integration/commands/actions/type_events_spec.js:595]: This line has been added not to allow more than 4 `li`s.

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js [hunk 3]
```diff
@@ -0,0 +1,770 @@
...
+      const targets = [
+        '#target-input-checkbox',
+        '#target-input-radio',
+      ]
+
+      targets.forEach((target) => {
+        it(target, () => {
+          cy.get(target).focus().type('{enter}')
+
+          cy.get('li').should('have.length', 3)
```
Comment (sainthkh) [inline packages/driver/cypress/integration/commands/actions/type_events_spec.js:614]: This line has been added not to allow more than 3 lis.

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js [hunk 4]
```diff
@@ -0,0 +1,770 @@
...
+          cy.get('li').eq(2).should('have.text', 'click')
+          cy.get('li').eq(3).should('have.text', 'keyup')
+        })
+      })
+    })
+
+    describe('does not trigger', () => {
+      const targets = [
+        '#target-input-checkbox',
+        '#target-input-radio',
```
Comment (sainthkh) [inline packages/driver/cypress/integration/commands/actions/type_events_spec.js:607]: I added `#target` to make tests consistent with `cy.type(' ')` tests.

**Review by sainthkh — COMMENTED** [review]
There are a few minor changes to the tests to make them consistent.

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js [hunk 5]
```diff
@@ -0,0 +1,770 @@
...
+
+        done()
+      })
+
+      cy.get(':text:first').type('a')
+    })
+
+    it('fires events in the correct order')
+
+    it('fires events for each key stroke')
```
Comment (emilyrohrbough) [inline packages/driver/cypress/integration/commands/actions/type_events_spec.js:139]: lets remove these empty tests or log an issue to back-fill these.

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js [hunk 6]
```diff
@@ -0,0 +1,770 @@
+const { _, $ } = Cypress
+
+describe('src/cy/commands/actions/type - #type events', () => {
+  beforeEach(() => {
+    cy.visit('fixtures/dom.html')
+  })
+
+  describe('events', () => {
```
Comment (emilyrohrbough) [inline packages/driver/cypress/integration/commands/actions/type_events_spec.js:8]: Would it be appropriate to rename this to `keyboard events`?

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js (hunk 5 above)
Comment (sainthkh) [reply packages/driver/cypress/integration/commands/actions/type_events_spec.js:139]: I tracked the history of `type_spec.js` and found out that it was like this from the beginning(6faba005c7feffa7ca3a387351647ade8b32b553). IMHO, logging an issue and implementing these tests would be a better idea. But I'm not sure why it is left empty.

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js [hunk 7]
```diff
@@ -0,0 +1,770 @@
...
+    })
+
+    it('fires when element is tabbed away from')//, ->
+    //   changed = 0
+
+    //   cy.$$(":text:first").change ->
+    //     changed += 1
+
+    //   cy.get(":text:first").invoke("val", "foo").type("b{tab}").then ->
+    //     expect(changed).to.eq 1
```
[the remark this answers is not in the document]
Comment (sainthkh) [reply packages/driver/cypress/integration/commands/actions/type_events_spec.js:358]: It is also like this long ago(d9bd8bf935c1f6ce3254264bd92169e95784c1f2, [link](https://github.com/cypress-io/cypress/commit/d9bd8bf935c1f6ce3254264bd92169e95784c1f2#r66773500)). 

It looks like a valid test to me. IMHO, adding this to the issue above is a good idea.

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js (hunk 6 above)
Comment (sainthkh) [reply packages/driver/cypress/integration/commands/actions/type_events_spec.js]: You're right. `keyboard events` better describes the tests. I fixed it that way.

File: packages/driver/cypress/integration/commands/actions/type_events_spec.js (hunk 5 above)
Comment (emilyrohrbough) [reply packages/driver/cypress/integration/commands/actions/type_events_spec.js:139]: Sounds like a good idea. Mind logging the issue & add TODO with link to these?

Comment (sainthkh) [reply packages/driver/cypress/integration/commands/actions/type_events_spec.js:139]: I opened this issue at #20283.

**Review by emilyrohrbough — APPROVED** [review]

**Review by BlueWinds — APPROVED** [review]

**Pull request merged**